phovea_security_store_generated/api.py

```diff
@@ -0,0 +1,34 @@
+from phovea_server.ns import Namespace
+from phovea_server.util import jsonify
+import logging
+import pydng
+
+_log = logging.getLogger(__name__)
+
+app = Namespace(__name__)
+
+
+def username_exists(username):
+  from phovea_server.security import manager
+  try:
+    user_id = manager()._load_user(username).get_id()
+    return username == user_id
+  except AttributeError:
+    return False
+
+
+@app.route('/generated_username')
+def get_generated_username():
+  from phovea_server.util import random_id
+
+  new_username = f"{pydng.generate_name()}_{random_id(1)}"
```

Comment:
The format is now `adjective_name_{x}` where x is a letter or number. The library https://pypi.org/project/pydng/ does nut support appending a number in the end. @thinkh Any better ideas?